src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets

```diff
@@ -129,6 +129,7 @@ The .NET Foundation licenses this file to you under the MIT license.
       <NativeLibrary Condition="'$(_targetArchitecture)' == 'x64'" Include="$(IlcSdkPath)$(VxSortSupportName)$(LibFileExt)" />
       <NativeLibrary Include="$(IlcSdkPath)$(StandaloneGCSupportName)$(LibFileExt)" />
       <NativeLibrary Condition="'$(LinkStandardCPlusPlusLibrary)' != 'true' and '$(StaticICULinking)' != 'true'" Include="$(IlcSdkPath)libstdc++compat.a" />
+      <NativeLibrary Include="$(IlcSdkPath)libz.a" />
```

Comment:
Native AOT is supported for mobile. We should use the same default zlib strategy for all runtime flavors targeting given platform. I think you should introduce a property like `UseSystemZLib` and use it in these target files.

(You will also want to run outer loop native aot tests to make sure that it works.)

Reply:
I assumed that you are planning to use system zlib for mobile and browser/wasm given the discussion in https://github.com/dotnet/runtime/issues/101465 , but it looks like that you are planning to use system zlib for browser wasm only. What is the plan?

In any case, it would be useful to have a property to disable use of zlib-ng for mobile and native aot that advanced users can use to resolve the conflicts. This property is also going to be useful to switch to system zlib in source build scenarios.

Reply:
At the moment of writing my comment in the main issue, I was testing using our copy of zlib-ng _everywhere_. Last night I was able to get it to work for all platforms (just one CI leg failing but I think I got it fixed just now).

Since this won't make it into preview6, I was thinking I could do the check for an installed system zlib in a separate PR, as we had originally discussed it initially when I created the PR:

![image](https://github.com/dotnet/runtime/assets/1175054/370532d8-9f12-425c-80d4-fb0b8823755f)

Thanks for your suggestions, I will make sure to keep them in mind for when I work on that.

Reply:
I decided to address all this in the current PR.